test/xpu_api/random/device_tests/bernoulli_distribution_device_test.pass.cpp

```diff
@@ -36,98 +35,178 @@ main()
 #if TEST_DPCPP_BACKEND_PRESENT && TEST_UNNAMED_LAMBDAS
 
     sycl::queue queue(exception_handler);
-    int err = 0;
-
-    // testing oneapi::dpl::bernoulli_distribution<sycl::vec<bool, 1>> oneapi::dpl::linear_congruential_engine
-    std::cout << "---------------------------------------------------------------------" << std::endl;
-    std::cout << "bernoulli_distribution<sycl::vec<bool, 1>> linear_congruential_engine" << std::endl;
-    std::cout << "---------------------------------------------------------------------" << std::endl;
-    err = device_copyable_test<oneapi::dpl::bernoulli_distribution<sycl::vec<bool, 1>>, oneapi::dpl::linear_congruential_engine<std::uint32_t,  a, c, m>>(queue);
-#if TEST_LONG_RUN
-    err += device_copyable_test<oneapi::dpl::bernoulli_distribution<sycl::vec<bool, 1>>, oneapi::dpl::linear_congruential_engine<sycl::vec<std::uint32_t, 16>,  a, c, m>>(queue);
-    err += device_copyable_test<oneapi::dpl::bernoulli_distribution<sycl::vec<bool, 1>>, oneapi::dpl::linear_congruential_engine<sycl::vec<std::uint32_t, 8>,  a, c, m>>(queue);
-    err += device_copyable_test<oneapi::dpl::bernoulli_distribution<sycl::vec<bool, 1>>, oneapi::dpl::linear_congruential_engine<sycl::vec<std::uint32_t, 4>,  a, c, m>>(queue);
-    err += device_copyable_test<oneapi::dpl::bernoulli_distribution<sycl::vec<bool, 1>>, oneapi::dpl::linear_congruential_engine<sycl::vec<std::uint32_t, 3>,  a, c, m>>(queue);
-    err += device_copyable_test<oneapi::dpl::bernoulli_distribution<sycl::vec<bool, 1>>, oneapi::dpl::linear_congruential_engine<sycl::vec<std::uint32_t, 2>,  a, c, m>>(queue);
-    err += device_copyable_test<oneapi::dpl::bernoulli_distribution<sycl::vec<bool, 1>>, oneapi::dpl::linear_congruential_engine<sycl::vec<std::uint32_t, 1>,  a, c, m>>(queue);
-#endif // TEST_LONG_RUN
-    EXPECT_TRUE(!err, "Test FAILED");
+    if (TestUtils::has_type_support<double>(queue.get_device()))

```

Comment:
double type is not used by the test. Does it mean that the RNG suite relies use it under the hood? If so, I believe it should be either fixed or documented.

Reply:
yes, it's used underneath. Good point to add to our documentation